server/view/view.controller.ts

```diff
@@ -104,4 +112,25 @@ export class ViewController {
             .getNextServer()
             .render(req, res, "/404-page", Object.assign(parsedUrl.query));
     }
+
+    @IsPublic()
+    @Get("unauthorized")
+    @Header("Cache-Control", "max-age=86400")
```

Comment:
If you want to use cached response, you shouldn't use variables in the Unauthorised message, like rolê and previous url.